### PR TITLE
[serve] Add warning for small smoothing factors.

### LINKED ISSUE #42342: [Serve] Print warnings when setting smoothing-factor =< 0.55
### What happened + What you expected to happen

Setting smoothing factor to 0.5, entails that to scale down from 2 replicas to 1 traffic has to literally be 0, which lead to a confusion by users not being able to understand why downscaling doesn't kick in.

One low-hanging fruit we can do in this case is to print warnings when users set values below 0.55 that their workload is unlikely to downscale below `1 / factor` value.

### Versions / Dependencies

2.9

### Reproduction script

Set `smoothing_factor=0.5`

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

A temporary mitigation to the issue before we find a more complete solution.

## Related issue number

Closes https://github.com/ray-project/ray/issues/42342.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/_private/deployment_state.py [hunk 1]
```diff
@@ -1516,6 +1516,18 @@ def deploy(self, deployment_info: DeploymentInfo) -> bool:
...
+            downscale_smoothing_factor = (
+                autoscaling_policy_manager.config.get_downscale_smoothing_factor()
+            )
+            if downscale_smoothing_factor <= 0.5:
+                logger.warning(
+                    "The downscale smoothing factor for deployment "
+                    f"'{self.deployment_name}' is set to a low value: "
+                    f"{downscale_smoothing_factor}. Note that this means the "
+                    "deployment is unlikely to scale below "
+                    f"{int(1 / downscale_smoothing_factor)} replicas."
```
Comment (edoakes) [inline python/ray/serve/_private/deployment_state.py:1529]:
```suggestion
                    "The downscale smoothing factor for deployment "
                    f"'{self.deployment_name}' is set to a low value: "
                    f"{downscale_smoothing_factor}. This means the "
                    "deployment is unlikely to scale below "
                    f"{int(1 / downscale_smoothing_factor)} replicas."
```

Comment (edoakes) [reply python/ray/serve/_private/deployment_state.py:1529]: Can we also add a minimum recommended smoothing factor to the error message?

Comment (zcin) [reply python/ray/serve/_private/deployment_state.py]: Sure, I put the recommended min as 0.6. I think anything above 0.5 is an OK recommendation, but any number between 0.5-0.6 seems kind of arbitrary.

**Review by edoakes — APPROVED** [review]

**Comment by zcin** [comment]
@edoakes Is this ready to merge?

**Pull request merged**